server/sonar-server/src/main/java/org/sonar/server/source/index/SourceFileResultSetIterator.java

```diff
@@ -79,23 +81,45 @@ public long getUpdatedAt() {
     "updated_at",
     "binary_data"
   };
-  private static final String SQL_ALL = "select " + StringUtils.join(FIELDS, ",") + " from file_sources";
-  private static final String SQL_AFTER_DATE = SQL_ALL + " where updated_at>?";
 
-  public static SourceFileResultSetIterator create(DbClient dbClient, Connection connection, long afterDate) {
+  private static final String SQL_ALL = "SELECT " + StringUtils.join(FIELDS, ",") + " FROM file_sources ";
+  private static final String AFTER_DATE_FILTER = "updated_at>? ";
+  private static final String PROJECT_FILTER = "project_uuid=? ";
+
+  public static SourceFileResultSetIterator create(DbClient dbClient, Connection connection, long afterDate, @Nullable String projectUuid) {
     try {
-      String sql = afterDate > 0L ? SQL_AFTER_DATE : SQL_ALL;
+      String sql = createSQL(afterDate, projectUuid);
       // rows are big, so they are scrolled once at a time (one row in memory at a time)
       PreparedStatement stmt = dbClient.newScrollingSingleRowSelectStatement(connection, sql);
       if (afterDate > 0L) {
         stmt.setLong(1, afterDate);
       }
+      if (projectUuid != null) {
+        stmt.setString(afterDate > 0L ? 2 : 1, projectUuid);
```

Comment:
you could use an index which starts at 0 and you put index++ each time it's used. WDYT ?
stmt.setLong(index++, afterDate); and stmt.setString(index++, projectUuid);